salt/transport/zeromq.py

```diff
@@ -69,33 +69,38 @@ def _get_master_uri(master_ip,
     '''
     Return the ZeroMQ URI to connect the Minion to the Master.
     It supports different source IP / port, given the ZeroMQ syntax:
-
     // Connecting using a IP address and bind to an IP address
     rc = zmq_connect(socket, "tcp://192.168.1.17:5555;192.168.1.1:5555"); assert (rc == 0);
-
     Source: http://api.zeromq.org/4-1:zmq-tcp
     '''
+    from salt.utils.zeromq import ip_bracket
+
     if LIBZMQ_VERSION_INFO >= (4, 1, 6) and ZMQ_VERSION_INFO >= (16, 0, 1):
         # The source:port syntax for ZeroMQ has been added in libzmq 4.1.6
         # which is included in the pyzmq wheels starting with 16.0.1.
         if source_ip or source_port:
             if source_ip and source_port:
                 return 'tcp://{source_ip}:{source_port};{master_ip}:{master_port}'.format(
-                        source_ip=source_ip, source_port=source_port,
-                        master_ip=master_ip, master_port=master_port)
+                        source_ip=ip_bracket(source_ip), source_port=source_port,
+                        master_ip=ip_bracket(master_ip), master_port=master_port)
             elif source_ip and not source_port:
                 return 'tcp://{source_ip}:0;{master_ip}:{master_port}'.format(
-                        source_ip=source_ip,
-                        master_ip=master_ip, master_port=master_port)
+                        source_ip=ip_bracket(source_ip),
+                        master_ip=ip_bracket(master_ip), master_port=master_port)
             elif not source_ip and source_port:
-                return 'tcp://0.0.0.0:{source_port};{master_ip}:{master_port}'.format(
-                        source_port=source_port,
+                master_ip_address = ipaddress.ip_address(master_ip)
+                if master_ip_address.Version == 4:
+                    ip_any = '0.0.0.0'
+                if master_ip_address.Version == 6:
```

Comment:
You should use `.version` property instead (lower-case). And since there is only two versions of IP, you can just use `else` instead. :wink: